packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -148,6 +148,11 @@ export function createOdspNetworkError(
         case fluidEpochMismatchError:
             error = new NonRetryableError(errorMessage, DriverErrorType.fileOverwrittenInStorage, { statusCode });
             break;
+        case 412:
+            // "Precondition Failed" error - happens when uploadSummaryWithContext uses wrong parent.
+            // Resubmitting same payload is not going to help, so this is non-recoverable failure!
+            error = new NonRetryableError(errorMessage, DriverErrorType.genericNetworkError, { statusCode });
```

Comment:
Is "genericNetworkError" the correct type? Should we have a different one?

Reply:
Maybe. There is no need for loader to differentiate here, and it never propagates to host, so realistically there is no need. As long as we have enough data to differentiate in the logs (which we do, as we report whole response).